### PR TITLE
Implement category list

### PR DESCRIPTION
**Pull request by Jifflis**
# 📲 What #
- Add category list api and test
 
# 🛠 How #
- Create resolver
- Create test
 
# ✅ Acceptance criteria #
- Category List 

# ⏰ TODO #
-none
 
# 🔗 References #
https://trello.com/c/nmSlMQ0G/18-list-categories-api

### CODE REVIEW

File: test/support/graphql/queries_helper.rb [hunk 1]
```diff
@@ -71,6 +71,34 @@ def category_groups_query(_input = {})
...
       )
     end
+
+    def categories_query(_input = {})
+      %(
+        query {
+          categories {
+            id
+            title,
+            budgets{
```
Comment (jaimejazarenoiii) [inline test/support/graphql/queries_helper.rb]: spacing before `{`

File: test/support/graphql/queries_helper.rb [hunk 2]
```diff
@@ -71,6 +71,34 @@ def category_groups_query(_input = {})
...
+              amount,
```
Comment (jaimejazarenoiii) [inline test/support/graphql/queries_helper.rb]: comma is not necessary in queries

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]

File: test/support/graphql/queries_helper.rb [hunk 3]
```diff
@@ -71,6 +71,34 @@ def category_groups_query(_input = {})
...
+        query {
+          categories {
+            id
+            title
+            budgets {
+              amount
+              id
+              createdAt
+              updatedAt
+            },
```
Comment (jaimejazarenoiii) [inline test/support/graphql/queries_helper.rb]: remove comma

File: test/graphql/queries/categories_query_test.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    test 'fetch categories' do
+      post('/graphql',
+           params: {
+             query: categories_query
+           }, headers: {
+             'Authorization': "Bearer #{@token}"
+           })
+      @json_response = parse_graphql_response(response.body)
+      assert_equal @json_response['categories'].count, 2
```
Comment (jaimejazarenoiii) [inline test/graphql/queries/categories_query_test.rb]: better to test the created categories(default after creation) so we know that the generation is done after creation of user

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]

**Review by rexendy — APPROVED** [review]

**Pull request merged**